app/helpers/sessions_helper.rb

```diff
@@ -0,0 +1,18 @@
+module SessionsHelper
+  def log_in user
+    session[:user_id] = user.id
+  end
+
+  def current_user
+    @current_user ||= User.find_by(id: session[:user_id])
```

Comment:
trong application controller include session helper này vào thì cái biến instance này dùng đc trong toàn bộ controller đó, function này kiểm tra biến đấy đã có  hay chưa, nếu chưa có thì find nó theo cái session lưu user_id, find_by nó tìm thì nếu thấy thì trả về user còn ko thì trả về nil thì current_user rỗng :v: